rust-src/bulletproofs/src/set_membership_proof.rs

```diff
@@ -0,0 +1,960 @@
+use crate::{inner_product_proof::*, utils::*};
+use crypto_common::*;
+use crypto_common_derive::*;
+use curve_arithmetic::{multiexp, multiexp_table, multiexp_worker_given_table, Curve};
+use ff::Field;
+use pedersen_scheme::*;
+use rand::*;
+use random_oracle::RandomOracle;
+use std::iter::once;
+
+#[derive(Clone, Serialize, SerdeBase16Serialize, Debug)]
+#[allow(non_snake_case)]
+pub struct SetMembershipProof<C: Curve> {
+    /// Commitment to the evaluation of the indicator function I_{v} on the_set
```

Comment:
```suggestion
    /// Commitment to the evaluation of the indicator function `I_{v}` on the_set
```
and similarly for the comments below. Make use of markdown.

In general I would also recommend that all these fields are either private or at most pub(crate). They are an implementation detail .